core/libp2p/multi/utils/base58_codec.cpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/multi/utils/base58_codec.hpp"
+
+#include <cstring>
+#include <vector>
+#include <cmath>
+
+#include "spdlog/spdlog.h"
+
+using kagome::common::Buffer;
+
+OUTCOME_CPP_DEFINE_CATEGORY(libp2p::multi, Base58Codec::DecodeError, e) {
+  using libp2p::multi::Base58Codec;  // not necessary, just for convenience
+  switch (e) {
+    case Base58Codec::DecodeError::kInvalidBase58Digit:
+      return "The digit is not used in base58 encoding";
+    case Base58Codec::DecodeError::kInvalidHighBit:
+      return "High-bit set on invalid digit";
+    case Base58Codec::DecodeError::kOutputTooBig:
+      return "Output number too big";
+    default:
+      return "Unknown error";
+  }
+}
+
+namespace libp2p::multi {
+
+  std::string Base58Codec::encode(std::string_view str) {
+    std::vector<uint8_t> v(str.begin(),
+                           str.end());  //< to convert from char* to uint8_t*
+                                        //without reinterpret cast
+    return Base58Codec::encode(v);
+  }
+
+  std::string Base58Codec::encode(gsl::span<const uint8_t> bytes) {
```

Comment:
Did you take it from somewhere? If so please provide a link to actual implementation